helidon-linker/src/main/java/io/helidon/linker/util/StreamUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.linker.util;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Stream utilities.
+ */
+public class StreamUtils {
+    private static final byte[] BUFFER = new byte[8196];
+
+    /**
+     * Transfers the contents of the given input stream to the given output stream.
+     *
+     * @param in The input stream.
+     * @param out The output stream.
+     * @throws IOException If an error occurs.
+     */
+    public static void transfer(InputStream in, OutputStream out) throws IOException {
+        synchronized (BUFFER) {
+            try (InputStream data = in) {
```

Comment:
There is `InputStream.transferTo(OutputStream)` method. Why not use that one?

Reply:
Oh, yes I can switch. (This changed over time and I didn't notice that it devolved into that same function.)

Reply:
Wait, I forgot... There is common path where this is used for every class in every jar, and this impl re-uses the buffer; this is why I kept it, to avoid all of the memory use and GC.

Reply:
In fact, looking at this now, I'm going to make the buffer much bigger ;-).